app/controllers/user_controller.php

```diff
@@ -2,37 +2,239 @@
+        }
+        if (self::MAX_POST_SIZE < $_SERVER['CONTENT_LENGTH']) {
+            $error = "Error! File Size too big! Can only upload 2 Mega Bytes";
+        }
+        if ($_FILES['avatar']['error'] === 4 && !isset($error)) {
+            $error = "No File was Selected";
+        }
+        if ($_FILES['avatar']['error'] > 0 && !isset($error)) {
+            $error = "Upload Error Occured";
+        }
+        if (isset($error)) {
+            $this->set(get_defined_vars());
+            return;
+        }
+        $file = $_FILES['avatar'];
+        $upload = new UploadImg($file);
+        if (!$upload->isFileAccepted()) {
+            $error = "File Upload Error";
+            $this->set(get_defined_vars());
+            return;
+        }
+        $upload->rename($user->username);
+        if (strpos($user->user_details['avatar'], 'default') === false) {
+            unlink(APP_DIR.'webroot'.$user->user_details['avatar']);
+        }
+        $saved = $upload->save('webroot/public_images/');
+        if (!$saved) {
+            $error = "Unexpected Error Occured";
+            $this->set(get_defined_vars());
+            return;
+        }
+        $user->saveAvatar($saved);
+        chmod(APP_DIR.'webroot'.$saved, 0755);
+        $this->set(get_defined_vars());
+    }
+
+    public function search()
+    {
+        $type = Param::get('type', false);
+        $query = Param::get('query', false);
+        if (!$type || !$query) {
+            redirect(url('user/profile'));
+        }
+        $user = new User($_SESSION['username']);
+        switch ($type) {
+            case ('user'):
+                $render_page = '/user/search';
+                $varname = 'query_results';
+                break;
+            case ('thread'):
+                $render_page = '/thread/index';
+                $varname = 'threads';
+                break;
+            case ('comment'):
+                
+                $render_page = '/comment/view';
+                $varname = 'comments';
+                break;
+            default:
+                redirect(url('user/profile'));
+                break;
+        }
+        $title = sprintf("Search for '%s' in %s", $query, $type);
+        $url_params = "?type={$type}&query={$query}";
+        $query_results = $user->search($type, $query);
+        $page = Param::get('page', 1);
+        $page = (int) $page;
+        $per_page = 5;
+        $pagination = new SimplePagination($page, $per_page);
+        $results = array();
+        for ($i = ($pagination->start_index - 1); $i < (($pagination->count)*$page); $i++) {
+            if (isset($query_results[$i])) {
+                $results[] = $query_results[$i];
+            }
         }
-        $home = '/user/index';
-        $user->username = $_SESSION['username'];
-        $user_id = Param::get('user_id',0);
-        $user_details = $user->getUserDetail($user_id);
+        $total = count($query_results);
+        ${$varname} = $results;
```

Comment:
I used that since I am using different views that expects different variable names that contains array of objects to be displayed, instead of creating new view, instead I used old views that are already existing so I need to pass the $results array to the proper variable name using $varname. Thanks :bow: